### PR TITLE
chore: Update electron to v28.1.4

### LINKED ISSUE #5285: Electron modules may not have ids now
### Bug description

in Electron 28 and support of ESM some modules may not have id/children properties

we should handle that else startup is failing with Electron 28

### Operating system

macOS 14

### Installation Method

None

### Version

next (development version)

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by benoitf**
### What does this PR do?
Includes a fixes for handling new module format in Electron

### What issues does this PR fix or reference?

fixes #5285 
https://github.com/containers/podman-desktop/pull/5124

### How to test this PR?

PR check should be ✅

### CODE REVIEW

File: packages/main/src/plugin/extension-loader.ts [hunk 1]
```diff
@@ -1092,18 +1096,23 @@ export class ExtensionLoader {
...
+  protected getRequireCache(): RequireCacheDict {
+    return require.cache;
+  }
+
   loadRuntime(extension: AnalyzedExtension): NodeRequire | undefined {
     // cleaning the cache for all files of that plug-in.
-    Object.keys(require.cache).forEach(function (key): void {
-      const mod: NodeJS.Module | undefined = require.cache[key];
+    Object.keys(this.getRequireCache()).forEach(key => {
+      const mod: NodeJS.Module | undefined = this.getRequireCache()[key];
```
Comment (dgolovin) [inline packages/main/src/plugin/extension-loader.ts]:
```suggestion
      const mod: NodeModule | undefined = this.requireCache[key];
```

File: packages/main/src/plugin/extension-loader.ts [hunk 2]
```diff
@@ -1092,18 +1096,23 @@ export class ExtensionLoader {
     return require(path);
   }
 
+  // helper function to get require cache
+  protected getRequireCache(): RequireCacheDict {
+    return require.cache;
+  }
```
Comment (dgolovin) [inline packages/main/src/plugin/extension-loader.ts]:
```suggestion
  // helper function to get require cache
  protected get requireCache(): RequireCacheDict {
    return require.cache;
  }
```

File: packages/main/src/plugin/extension-loader.ts [hunk 3]
```diff
@@ -1092,18 +1096,23 @@ export class ExtensionLoader {
...
+
   loadRuntime(extension: AnalyzedExtension): NodeRequire | undefined {
     // cleaning the cache for all files of that plug-in.
-    Object.keys(require.cache).forEach(function (key): void {
-      const mod: NodeJS.Module | undefined = require.cache[key];
+    Object.keys(this.getRequireCache()).forEach(key => {
```
Comment (dgolovin) [inline packages/main/src/plugin/extension-loader.ts]:
```suggestion
    Object.keys(this.requireCache).forEach(key => {
```

File: packages/main/src/plugin/extension-loader.ts (hunk 1 above)
Comment (dgolovin) [reply packages/main/src/plugin/extension-loader.ts]: Would be easier to read.

**Review by dgolovin — COMMENTED** [review]
Some nitpicking

File: packages/main/src/plugin/extension-loader.ts (hunk 1 above)
Comment (benoitf) [reply packages/main/src/plugin/extension-loader.ts]: fixed

**Comment by benoitf** [comment]
new playwright fixed the issue with electron v28 so now e2e tests can run

**Review by deboer-tim — APPROVED** [review]
Code LGTM, tested for several minutes and couldn't see any difference/no problems.

**Pull request merged**